resolwe/flow/managers/workload_connectors/kubernetes.py

```diff
@@ -860,6 +863,32 @@ def start(self, data: Data, listener_connection: Tuple[str, str, str]):
             "It took {:.2f}s to send config to kubernetes".format(end_time - start_time)
         )
 
+        # Patch the PVC-s with owner references to the job.
+        patch = []
+        for claim_name in created_claim_names:
+            patch.append(
+                {
+                    "op": "add",
+                    "path": "/metadata/ownerReferences",
+                    "value": [
+                        {
+                            "apiVersion": "batch/v1",
+                            "kind": "Job",
+                            "name": job.metadata.name,
+                            "uid": job.metadata.uid,
+                        }
+                    ],
+                }
+            )
+
+        if patch:
+            # Patch the PVC with ownerReference. This ensures the PVC claim gets deleted
+            # when the job has completed or failed. The PVCs are deleted after
+            # ttlSecondsAfterFinished seconds, declared in Job spec.
+            core_api.patch_namespaced_persistent_volume_claim(
```

Comment:
Patch is a list, containing typically one patch for each claim. I think you need to add iteration over all patches.

Reply:
Done.